bin/check_smoker_plugin.py

```diff
@@ -22,51 +26,95 @@
+                            "nor --health specified")
 
     # No plugin found
     if not plugins:
-        nagios.exit_unknown("plugin %s not found" % args.plugin)
+        if args.plugin:
+            nagios.exit_unknown("plugin %s not found" % args.plugin)
+        else:
+            nagios.exit_ok("no plugin found by category %s and component %s, health: %s"
+                           % (args.category, args.component, args.health))
 
     if args.force:
         plugins = client.force_run(plugins, progress=False)
 
+    status_methods_pairs = [(ERROR, nagios.exit_critical), (UNKNOWN, nagios.exit_unknown),
+                            (WARN, nagios.exit_warning), (OK, nagios.exit_ok)]
+
     # Manage plugin result. We can't return much data to Nagios, so just say if it's alright or not
+    results = dict((s, []) for s, _ in status_methods_pairs)
+
     for plugin in plugins.values()[0]['plugins'].itervalues():
+        plugin_name = plugin['name']
         if not plugin['lastResult']:
-            # No result?! -> UNKNOWN
-            nagios.exit_unknown("plugin has no last result")
-        elif plugin['lastResult']['status'] == 'OK':
-            try:
-                nagios.exit_ok(plugin['lastResult']['messages']['ok'][0])
-            except (KeyError, IndexError, TypeError):
-                nagios.exit_ok("smoke test %s succeeded at %s" % ( plugin['name'], plugin['lastResult']['lastRun']) )
-        elif plugin['lastResult']['status'] == 'WARN':
-            try:
-                nagios.exit_warning(plugin['lastResult']['messages']['warn'][0])
-            except (KeyError, IndexError, TypeError):
-                nagios.exit_warning(ParseIssue(plugin))
-        elif plugin['lastResult']['status'] == 'ERROR':
-            try:
-                nagios.exit_critical(plugin['lastResult']['messages']['error'][0])
-            except (KeyError, IndexError, TypeError):
-                nagios.exit_critical(ParseIssue(plugin))
+            results[UNKNOWN].append({'name': plugin_name, 'message': "plugin has no last result"})
         else:
-            nagios.exit_unknown("unknown status %s at %s" % (plugin['lastResult']['status'], plugin['lastResult']['lastRun']))
+            last_status = plugin['lastResult']['status']
+
+            if last_status in [ERROR, WARN, OK]:
+                try:
+                    results[last_status].append({'name': plugin_name,
+                                                 'message': plugin['lastResult']['messages'][last_status.lower()][0]})
+                except (KeyError, IndexError, TypeError):
+                    if last_status == OK:
+                        results[last_status].append({'name': plugin_name,
+                                                     'message': "smoke test %s succeeded at %s"
+                                                                % (plugin['name'], plugin['lastResult']['lastRun'])})
```

Comment:
this indent is ugly and not PEP8 compliant. The whole file is not compliant, but we should at least try to fix it in new code. How about 
```python
results[last_status].append(
    {'name': plugin_name,
    'message': "smoke test %s succeeded at %s" %
    (plugin['name'], plugin['lastResult']['lastRun'])})